Model/reviews.patch.model.js

```diff
@@ -0,0 +1,48 @@
+const db = require("../db/connection.js");
+
+exports.selectReviewPost = (review_id, patchInfo) => {
+  const id = review_id.reviews_id;
+  const toPatchThisKey = Object.keys(patchInfo);
+  const validReviewKeys = [
+    "title",
+    "review_body",
+    "designer",
+    "review_img_url",
+    "votes",
+    "category",
+    "owner",
+  ];
+  return db
+    .query(`SELECT * FROM reviews WHERE review_id = ${id};`)
+    .then((reviewsIdData) => {
+      if (reviewsIdData.rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Not Found" });
+      }
+      let currentValue = reviewsIdData.rows[0][toPatchThisKey[0]];
+      let queryStr = "UPDATE reviews";
+      let newValue = 0;
+      const createdQueryObj = {};
+      if (
+        typeof patchInfo[toPatchThisKey[0]] === "number" &&
+        toPatchThisKey[0] === "votes"
+      ) {
+        newValue = currentValue += patchInfo[toPatchThisKey[0]];
+      } else {
+        return Promise.reject({ status: 400, msg: "Bad Request" });
+      }
+      if (validReviewKeys.includes(toPatchThisKey[0])) {
+        queryStr += ` SET ${toPatchThisKey[0]} = $1`;
+        queryStr += ` WHERE review_id = $2 RETURNING *;`;
+      }
+      createdQueryObj.string = queryStr;
+      createdQueryObj.value = newValue;
+      createdQueryObj.id = id;
+      return createdQueryObj;
+    })
```

Comment:
We dont really need to do a check for the article first as we can simply do an UPDATE operation with a RETURNING and then check how many rows are returned to us. If we get rows.length === 0, then we know the article doesnt exist. This saves us doing 2 queries for this. Also passing the wrong data type into and SQL string will make SQL throw an error based on that so we dont really need to do a lot of this checking in JS

Reply:
ToDo less bulk and look for a sql code of 'abc' then let the custom error handling the rest...